gather.go

```diff
@@ -244,28 +252,35 @@ func (a *Agent) gatherCandidatesSrflx(urls []*URL, networkTypes []NetworkType) {
 						a.log.Warnf("Failed to create server reflexive candidate: %s %s %d: %v\n", network, ip, port, err)
 						return
 					}
-
-					if err := a.run(func(agent *Agent) {
-						a.addCandidate(c)
-					}); err != nil {
-						a.log.Warnf("Failed to append to localCandidates: %v\n", err)
-						return
+					select {
+					case serverReflexiveCandidate <- gatherResult{c, conn}:
```

Comment:
How do you know that the first is the default? I think when it returns local interfaces, they are just sorted by interface name. At least that is how `ifconfig` does it.

Reply:
There doesn't seem to be a portable way to get the `default` route so couldn't go that route.

Instead for now I say we just take the Srflx candidate that works. If people want we can give them a flag to `wait for all` 

But this way we still respect the port restriction, and we can get a candidate.